invenio_checks/components.py

```diff
@@ -21,31 +24,58 @@ def enabled(self):
         """Return if checks are enabled."""
         return current_app.config.get("CHECKS_ENABLED", False)
 
-    def update_draft(self, identity, data=None, record=None, errors=None, **kwargs):
+    def _run_checks(self, identity, data=None, record=None, errors=None, **kwargs):
         """Update handler."""
         errors = errors or []
 
         if not self.enabled:
             return
 
         communities = []
-        if record.review.is_open:
-            communities.append(record.review.community)
+        if record.parent.review and (
+            record.parent.review.status == "submitted"
+            or record.parent.review.status == "created"
+        ):
+            # drafts can only be submitted to one community
+            communities.append(record.parent.review.receiver.resolve().id)
+        else:
+            results = current_requests_service.search(
+                identity,
+                extra_filter=dsl.query.Bool(
+                    "must",
+                    must=[
+                        dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+                        dsl.Q("term", **{"is_open": True}),
+                    ],
+                ),
+            )
 
-        communities.extend(record.parent.communities.entries)
-        checks = itertools.chain(*[c.checks for c in communities])
+            if results.total > 0:
+                for result in results:
+                    communities.append(result.get("receiver", {}).get("community"))
+            else:
+                return
 
-        for check in checks:
+        all_checks = [
+            (CheckConfig.query.filter(CheckConfig.community_id == c).all(), c)
+            for c in communities
+        ]
+
+        for checks, community in all_checks:
             try:
-                res = check.run(data, record)
-                if not res.sync:
-                    continue
-                for error in res.errors:
-                    errors.append(error)
+                for check in checks:
+                    check_cls = current_checks_registry.get(check.check_id)
+                    res = check_cls().run(record, check.params, community)
+                    if not res.sync:
```

Comment:
I am not sure that I understand correctly this line: if the check has been already executed, why do we check the `sync` and `continue` after the execution?

Reply:
The logic is that if a check needs to be run asynchronously (i.e. via a Celery task), there won't be any errors returned that can be added to the rest.

At the moment we don't have any async checks implemented. Virus check, 3rd-party FAIR Score, etc. would be example async checks, but the entire model of where these show up, given that they take longer, is going to be focused around the request page, and not so much the deposit form.